recursive_art.py

```diff
@@ -20,8 +25,31 @@ def build_random_function(min_depth, max_depth):
         (See the assignment writ-eup for details on the representation of
         these functions)
     """
-    # TODO: implement this
-    pass
+    random_list = ["prod", "avg", "cos_pi", "sin_pi"]
+    random_func = random.choice(random_list)
+    if max_depth > 0:
```

Comment:
This is a giant conditional. There's nothing necessarily "wrong" with nesting conditionals this deep, but I've found that once you reach three layers of if statements it's probably best to sit back and try to understand why you need to do that, because chances are you can refactor it to be more readable. Adding comments explaining what you are doing helps as well.